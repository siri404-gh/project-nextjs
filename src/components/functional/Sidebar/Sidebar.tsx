import React, { FunctionComponent } from 'react';
import {
  SwipeableDrawer,
  Typography,
  makeStyles,
  Toolbar,
} from '@material-ui/core';

import Logobutton from '@/components/functional/Logobutton/Logobutton';

export interface Props {
  isOpen?: boolean,
  onOpen?: () => void,
  onClose?: () => void,
  title?: string,
  children?: JSX.Element,
}

const useStyles = makeStyles(theme => ({
  paper: {
    width: 285,
    '&::-webkit-scrollbar': {
      display: 'none',
    },
  },
  text: {
    padding: 5,
  },
  logoIcon: {
    height: 30,
    width: 30,
    padding: 6,
  },
  toolbarWrapper: {
    borderBottom: 'solid 1px #e4e4e4',
    position: 'sticky',
    top: 0,
    zIndex: theme.zIndex.drawer + 1,
    backgroundColor: '#ffffff',
  },
  toolbar: {
    minHeight: 56,
    paddingLeft: 10,
    paddingRight: 10,
  },
}));

const Sidebar: FunctionComponent<Props> = ({ children, isOpen, onOpen, onClose, title }) => {
  const classes = useStyles();

  return <SwipeableDrawer
    className='Sidebar'
    variant='temporary'
    swipeAreaWidth={50}
    anchor='left'
    classes={{ paper: classes.paper }}
    open={isOpen}
    onOpen={onOpen}
    onClose={onClose}>
    <div className={classes.toolbarWrapper}>
      <Toolbar className={classes.toolbar}>
        <Logobutton />
        <Typography variant='h6' color='textSecondary' className={classes.text} noWrap>{title}</Typography>
      </Toolbar>
    </div>
    {children}
  </SwipeableDrawer>;
};

Sidebar.defaultProps = {
  children: null,
  isOpen: false,
  title: '',
  onOpen: /* istanbul ignore next */ () => false,
  onClose: /* istanbul ignore next */ () => false,
};

export default Sidebar;
