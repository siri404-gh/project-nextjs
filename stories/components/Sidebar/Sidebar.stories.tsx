import React from 'react';
import { Story } from '@storybook/react/types-6-0';

import Sidebar, { Props } from '@/components/functional/Sidebar/Sidebar';

export default {
  title: 'Sidebar',
  component: Sidebar,
};

const Template: Story<Props> = args => <Sidebar {...args} />;

export const FirstStory = Template.bind({});

FirstStory.args = {
  isOpen: true,
  onClose: () => alert('close')
};
