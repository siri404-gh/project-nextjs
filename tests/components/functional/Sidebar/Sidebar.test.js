import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Sidebar from '@/components/functional/Sidebar/Sidebar';

describe('Sidebar', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('section');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render without props', () => {
    act(() => {
      render(<Sidebar />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
