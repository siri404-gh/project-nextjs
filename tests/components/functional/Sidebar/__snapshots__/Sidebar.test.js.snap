// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should render without props 1`] = `
<section>
  <div
    class="PrivateSwipeArea-root-6 PrivateSwipeArea-anchorLeft-7"
    style="width: 20px;"
  />
</section>
`;
